cpp/betweenness_centrality_module/betweenness_centrality_online_module.cpp

```diff
@@ -0,0 +1,267 @@
+          const std::pair<std::uint64_t, std::uint64_t> edge{graph->GetMemgraphNodeId(edge_inner_ids.from),
+                                                             graph->GetMemgraphNodeId(edge_inner_ids.to)};
+          // Newly created edges aren’t part of the prior graph
+          if (std::find(created_edges.begin(), created_edges.end(), edge) == created_edges.end())
+            prior_edges_ids.push_back(edge);
+        }
+        for (const auto deleted_edge : deleted_edges) {
+          prior_edges_ids.push_back(deleted_edge);
+        }
+
+        auto prior_graph =
+            mg_generate::BuildGraph(graph_nodes_ids, prior_edges_ids, mg_graph::GraphType::kUndirectedGraph);
+
+        node_bc_scores = algorithm.Set(*graph, normalize, threads);
+
+        for (const auto created_edge : created_edges) {
+          prior_edges_ids.push_back(created_edge);
+          graph = mg_generate::BuildGraph(graph_nodes_ids, prior_edges_ids, mg_graph::GraphType::kUndirectedGraph);
+          node_bc_scores = algorithm.EdgeUpdate(*prior_graph, *graph, online_bc::Operation::CREATE_EDGE, created_edge,
+                                                normalize, threads);
+          prior_graph = std::move(graph);
+        }
+        for (const auto deleted_edge : deleted_edges) {
+          prior_edges_ids.erase(std::remove(prior_edges_ids.begin(), prior_edges_ids.end(), deleted_edge),
+                                prior_edges_ids.end());
+          graph = mg_generate::BuildGraph(graph_nodes_ids, prior_edges_ids, mg_graph::GraphType::kUndirectedGraph);
+          node_bc_scores = algorithm.EdgeUpdate(*prior_graph, *graph, online_bc::Operation::DELETE_EDGE, deleted_edge,
+                                                normalize, threads);
+          prior_graph = std::move(graph);
+        }
+      } else if (created_edges.size() == 0 && deleted_edges.size() == 0) {  // Node update
+        for (const auto created_node_id : created_nodes) {
+          algorithm.NodeUpdate(online_bc::Operation::CREATE_NODE, created_node_id, normalize);
+        }
+        for (const auto deleted_node_id : deleted_nodes) {
+          algorithm.NodeUpdate(online_bc::Operation::DELETE_NODE, deleted_node_id, normalize);
+        }
+      } else if (created_nodes.size() == 1 && created_edges.size() == 1 &&
+                 ConnectedVia(created_nodes[0], created_edges[0])) {  // Create-and-attach-node update
+        algorithm.NodeEdgeUpdate(*graph, online_bc::Operation::CREATE_ATTACH_NODE, created_nodes[0], created_edges[0],
+                                 normalize);
+      } else if (deleted_nodes.size() == 1 && deleted_edges.size() == 1 &&
+                 ConnectedVia(deleted_nodes[0], deleted_edges[0])) {  // Detach-and-delete-node update
+        algorithm.NodeEdgeUpdate(*graph, online_bc::Operation::DETACH_DELETE_NODE, deleted_nodes[0], deleted_edges[0],
+                                 normalize);
```

Comment:
Could you repeat why having node creation/deletion with only 1 affected edge needs separate case

Reply:
That is the case which can be handled by a single iteration of the Brandes algorithm; as that only takes linear time, I believe adding it as an edge case is useful.

Reply:
That makes sense :+1: